lib/matplotlib/backends/backend_qt4.py

```diff
@@ -1,48 +1,16 @@
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
 
-import six
-from six import unichr
-import os
-import re
-import signal
-import sys
-
-import matplotlib
-
-from matplotlib.cbook import is_string_like
-from matplotlib.backend_bases import FigureManagerBase
 from matplotlib.backend_bases import FigureCanvasBase
-from matplotlib.backend_bases import NavigationToolbar2
-
-from matplotlib.backend_bases import cursors
-from matplotlib.backend_bases import TimerBase
-from matplotlib.backend_bases import ShowBase
-
-from matplotlib._pylab_helpers import Gcf
 from matplotlib.figure import Figure
 
+from .qt_compat import QtCore, QtWidgets, __version__
 
-from matplotlib.widgets import SubplotTool
-try:
-    import matplotlib.backends.qt_editor.figureoptions as figureoptions
-except ImportError:
-    figureoptions = None
-
-from .qt_compat import QtCore, QtWidgets, _getSaveFileName, __version__
-from matplotlib.backends.qt_editor.formsubplottool import UiSubplotTool
-
-from .backend_qt5 import (backend_version, SPECIAL_KEYS, SUPER, ALT, CTRL,
```

Comment:
@efiring @mdboom thoughts?  This issue is why I have thought about but
never done this pruning.

At a minimum all the removals need to be documented as api changes with
where to import them from instead.

The gui backends are heavily used by people embedding in larger
applications so many strange things may be going on.

The problem is that I don't think exactly what is part of the public api
has been well documented, anything with out a leading _ is considered
public in python and we err on the side of not being user code unless there
is a very compelling reason.

On Mon, Apr 20, 2015, 17:05 Kelvin Li <notifications@github.com> wrote:

> In lib/matplotlib/backends/backend_qt4.py
> <https://github.com/matplotlib/matplotlib/pull/4363#discussion_r28729127>:
>
> >
> > -from matplotlib.widgets import SubplotTool
> > -try:
> > -    import matplotlib.backends.qt_editor.figureoptions as figureoptions
> > -except ImportError:
> > -    figureoptions = None
> > -
> > -from .qt_compat import QtCore, QtWidgets, _getSaveFileName, __version__
> > -from matplotlib.backends.qt_editor.formsubplottool import UiSubplotTool
> > -
> > -from .backend_qt5 import (backend_version, SPECIAL_KEYS, SUPER, ALT, CTRL,
>
> Thanks for the tips. Just for my own understanding, does this mean we
> cannot break users who do the following?
>
> from matplotlib.backends.backend_qt4 import SubplotTool
>
> I'm surprised that this is considered part of the external API, but I'll
> take your word for it.
>
> Assuming we are on the same page, then moving forward, I'm not even sure
> what is considered safe to prune... any advice?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4363/files#r28729127>.
>